src/tensor/index.rs

```diff
@@ -0,0 +1,304 @@
+        self.indexer(&[index.into()])
+    }
+}
+
+impl<A> IndexOp<(A,)> for Tensor
+where
+    A: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A,)) -> Tensor {
+        let a = index.0.into();
+        self.indexer(&[a])
+    }
+}
+
+impl<A, B> IndexOp<(A, B)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        self.indexer(&[a, b])
+    }
+}
+
+impl<A, B, C> IndexOp<(A, B, C)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+    C: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B, C)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        let c = index.2.into();
+        self.indexer(&[a, b, c])
+    }
+}
+
+impl<A, B, C, D> IndexOp<(A, B, C, D)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+    C: Into<TensorIndexer>,
+    D: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B, C, D)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        let c = index.2.into();
+        let d = index.3.into();
+        self.indexer(&[a, b, c, d])
+    }
+}
+
+impl<A, B, C, D, E> IndexOp<(A, B, C, D, E)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+    C: Into<TensorIndexer>,
+    D: Into<TensorIndexer>,
+    E: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B, C, D, E)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        let c = index.2.into();
+        let d = index.3.into();
+        let e = index.4.into();
+        self.indexer(&[a, b, c, d, e])
+    }
+}
+
+impl<A, B, C, D, E, F> IndexOp<(A, B, C, D, E, F)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+    C: Into<TensorIndexer>,
+    D: Into<TensorIndexer>,
+    E: Into<TensorIndexer>,
+    F: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B, C, D, E, F)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        let c = index.2.into();
+        let d = index.3.into();
+        let e = index.4.into();
+        let f = index.5.into();
+        self.indexer(&[a, b, c, d, e, f])
+    }
+}
+
+impl<A, B, C, D, E, F, G> IndexOp<(A, B, C, D, E, F, G)> for Tensor
+where
+    A: Into<TensorIndexer>,
+    B: Into<TensorIndexer>,
+    C: Into<TensorIndexer>,
+    D: Into<TensorIndexer>,
+    E: Into<TensorIndexer>,
+    F: Into<TensorIndexer>,
+    G: Into<TensorIndexer>,
+{
+    fn i(&self, index: (A, B, C, D, E, F, G)) -> Tensor {
+        let a = index.0.into();
+        let b = index.1.into();
+        let c = index.2.into();
+        let d = index.3.into();
+        let e = index.4.into();
+        let f = index.5.into();
+        let g = index.6.into();
+        self.indexer(&[a, b, c, d, e, f, g])
+    }
+}
+
+impl Tensor {
+    fn indexer(&self, index_spec: &[TensorIndexer]) -> Tensor {
```

Comment:
Similar to other tensor function, can you write a fallible version `f_indexer(...)` and have `indexer(...)` call it + `unwrap` ?